src/topology/algebra/uniform_field.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2019 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import topology.algebra.uniform_ring
+import topology.algebra.field
+
+/-!
+# Completion of topological fields
+
+The goal of this file is to prove the main part of Proposition 7 of Bourbaki GT III 6.8 :
+
+The completion `hat K` of a Hausdorff topological field is a field if the image under
+the mapping `x ↦ x⁻¹` of every Cauchy filter (with respect to the additive uniform structure)
+which does not have a cluster point at `0` is a Cauchy filter
+(with respect to the additive uniform structure).
+
+Bourbaki does not give any detail here, he refers to the general discussion of extending
+functions defined on a dense subset with values in a complete Hausdorff space. In particular
+the subtlety about clustering at zero is totally left to readers.
+
+Note that the separated completion of a non-separated topological field is the zero ring, hence
+the separation assumption is needed. Indeed the kernel of the completion map is the closure of
+zero which is an ideal. Hence it's either zero (and the field is separated) or the full field,
+which implies one is sent to zero and the completion ring is trivial.
+
+The main definition is `completable_top_field` which packages the assumptions as a Prop-valued
+type class and the main results are the instances `field_completion` and
+`topological_division_ring_completion`.
+-/
+
+
+noncomputable theory
+
+open_locale classical uniformity topological_space
+
+open set uniform_space uniform_space.completion filter
+
+variables (K : Type*) [field K]  [uniform_space K]
+
+local notation `hat` := completion
```

Comment:
At the moment you have this weirdness where all the lemmas use `hat` in their name, but the name is nowhere to be seen in the goal view when looking for these lemmas as the notation is local.

I'd suggest either:
* Changing this to `abbreviation` so that the name sticks around
* Renaming the lemmas to continue to use `completion`

Reply:
Where do you see a lemma using hat in their name? There are lemmas using `hat_inv` in their name, but this is not a notation, that's an actual declaration name.

Reply:
You're quite right, my mistake.